luminoth/models/fasterrcnn/rpn_proposal.py

```diff
@@ -118,6 +117,14 @@ def _build(self, rpn_cls_prob, rpn_bbox_pred, all_anchors, im_shape):
             all_proposals, proposal_filter,
             name='filter_invalid_proposals'
         )
+        if self._debug:
+            proposals_raw = tf.identity(proposals)
+            scores_raw = tf.identity(scores)
+
+        if not self._clip_after_nms:
+            # Clip proposals to the image.
+            proposals = clip_boxes(proposals, im_shape)
```

Comment:
Why move clip to after filtering zero area proposals?

Reply:
I moved that because if not, in case you clip the proposals before nms you are processing all the invalid area proposals, which I think is unnecessary and not efficient.
Also it allows the creation of the tensor proposals_raw (now to be proposals_unfiltered) which doesn't include the invalid area proposals but aren't filtered.